src/modules/cpu/kernel/ricap.hpp

```diff
@@ -0,0 +1,1572 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_simd.hpp"
+#include "rpp_cpu_common.hpp"
+
+    // RICAP output image profile as per "Data Augmentation using Random Image Cropping and Patching for Deep CNNs" available at https://arxiv.org/pdf/1811.09030.pdf
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-1------|------img-roi-2------|
+    // |------img-roi-3------|------img-roi-4------|
+    // |------img-roi-3------|------img-roi-4------|
+    // |------img-roi-3------|------img-roi-4------|
+
+RppStatus ricap_u8_u8_host_tensor(Rpp8u *srcPtr,
```

Comment:
it is hard to verify the algorithm correctness by reviewing this code.
Is there a unit-test which runs the algorithm in C code  and compares the output with this?

Reply:
Yes Rajy, there's support for QA test and unit-tests to visually verify the outputs.

Reply:
We have compared outputs from both C and assembly codes during development and they match pixelwise.